R/correlogram.R

```diff
@@ -0,0 +1,214 @@
+#' Create a correlogram from a given dataframe
+#'
+#' @param df A dataframe containing desired variables. Can be
+#' inputted as continuous, binary, or ordinal variables.
+#' Note: Binary variables must have a value of 0 or 1.
+#' Note: Ordinal variables must be formatted as factors.
+#' @param fig_colors Allows the user to change the colors of the figure
+#' (defaults are provided). Must be vector of length 3, with color corresponding
+#' to strength of correlation.
+#' @param diagonal Allows user to choose to view the correlogram with diagonal
+#' entries (default is FALSE).
+#'
+#' @return A correlogram.
+#' @export
+#' @import ltm
+#' @import tibble
+#' @import rcompanion
+#' @import cowplot
+#' @import ggcorrplot
+#' @import stringr
+#' @import faux
+#'
+#' @details
+#' Different correlation coefficients are calculated based on the nature of the
+#' variables:
+#' For two continuous variables, the Pearson correlation coefficient is used.
+#' For two binary variables, Cramér's phi association is implemented.
+#' For one binary and one continuous variable, point biserial correlation is
+#' utilized.
+#' For two ordinal variables, Spearman rank correlation is utilized.
+#' For one continuous and one ordinal variable, a modified Pearson correlation
+#' combined with the nonparametric Spearman rank correlation is used.
+#' For one binary and one ordinal variable, Glass rank biserial correlation
+#' is implemented.
+#'
+#' @examples
+#' create_correlogram(corr)
+create_correlogram <- function(df,
+                               diagonal = FALSE,
+                               fig_colors = colfun()$fig10_colors) {
```

Comment:
Consider adding these as parameters
```
method = "square",
  type = "lower"
```

Reply:
Hi LG, thanks for the suggestion! I was able to implement it within the function. Let me know if you have additional ideas!

Reply:
Thanks Abigail - looks good!